File: assignments/assignment-2-solution/src-csharp/TeaShop.UnitTest/Domain/PaymentStrategy/PaymentStrategyTests.cs
using Assignment2Solution.Domain.Inventory;
using Assignment2Solution.Domain.PaymentStrategy;

namespace TeaShop.UnitTest.Domain.PaymentStrategy;

public class PaymentStrategyTests
{
    private readonly InventoryItem _testItem = new(Guid.NewGuid(), "Green Tea", 15.00m, 50, new StarRating(4));

    [Fact]
    public void ApplePayStrategy_Checkout_WritesToOutput()
    {
        // Arrange
        var strategy = new ApplePayStrategy("user@example.com");
        using var writer = new StringWriter();

        // Act
        strategy.Checkout(_testItem, 2, writer);

        // Assert
        var output = writer.ToString();
        Assert.Contains("Apple Pay", output);
        Assert.Contains("user@example.com", output);
        Assert.Contains("$30.00", output);
    }

    [Fact]
    public void ApplePayStrategy_EmptyUsername_ThrowsException()
    {
        // Act & Assert
        Assert.Throws<ArgumentException>(() => new ApplePayStrategy(""));
    }

    [Fact]
    public void CreditCardStrategy_Checkout_WritesToOutput()
    {
        // Arrange
        var strategy = new CreditCardStrategy("1234567890123456");
        using var writer = new StringWriter();

        // Act
        strategy.Checkout(_testItem, 2, writer);

        // Assert
        var output = writer.ToString();
        Assert.Contains("Credit Card", output);
        Assert.Contains("1234567890123456", output);
        Assert.Contains("$30.00", output);
    }

    [Fact]
    public void CryptoCurrencyStrategy_Checkout_WritesToOutput()
    {
        // Arrange
        var strategy = new CryptoCurrencyStrategy("0x123abc456def", "sig123");
        using var writer = new StringWriter();

        // Act
        strategy.Checkout(_testItem, 2, writer);

        // Assert
        var output = writer.ToString();
        Assert.Contains("Crypto", output);
        Assert.Contains("0x123abc456def", output);
        Assert.Contains("sig123", output);
        Assert.Contains("$30.00", output);
    }
}

File: assignments/assignment-2-solution/src-java/src/test/java/teashop/unittest/domain/paymentstrategy/PaymentStrategyTests.java
package teashop.unittest.domain.paymentstrategy;

import assignment2solution.domain.inventory.InventoryItem;
import assignment2solution.domain.inventory.StarRating;
import assignment2solution.domain.payment.ApplePayStrategy;
import assignment2solution.domain.payment.CreditCardStrategy;
import assignment2solution.domain.payment.CryptoCurrencyStrategy;

import org.junit.jupiter.api.Test;

import java.io.StringWriter;
import java.math.BigDecimal;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.*;

public class PaymentStrategyTests {
    private final InventoryItem testItem = new InventoryItem(
        UUID.randomUUID(),
        "Green Tea",
        new BigDecimal("15.00"),
        50,
        new StarRating(4)
    );

    @Test
    public void applePayStrategyCheckoutWritesToOutput() {
        var strategy = new ApplePayStrategy("user@example.com");
        var writer = new StringWriter();

        strategy.checkout(testItem, 2, writer);

        var output = writer.toString();
        assertTrue(output.contains("Apple Pay"));
        assertTrue(output.contains("user@example.com"));
        assertTrue(output.contains("$30.00"));
    }

    @Test
    public void applePayStrategyEmptyUsernameThrowsException() {
        assertThrows(IllegalArgumentException.class, () -> new ApplePayStrategy(""));
    }

    @Test
    public void creditCardStrategyCheckoutWritesToOutput() {
        var strategy = new CreditCardStrategy("1234567890123456");
        var writer = new StringWriter();

        strategy.checkout(testItem, 2, writer);

        var output = writer.toString();
        assertTrue(output.contains("Credit Card"));
        assertTrue(output.contains("1234567890123456"));
        assertTrue(output.contains("$30.00"));
    }

    @Test
    public void cryptoCurrencyStrategyCheckoutWritesToOutput() {
        var strategy = new CryptoCurrencyStrategy("0x123abc456def", "sig123");
        var writer = new StringWriter();

        strategy.checkout(testItem, 2, writer);

        var output = writer.toString();
        assertTrue(output.contains("Crypto"));
        assertTrue(output.contains("0x123abc456def"));
        assertTrue(output.contains("sig123"));
        assertTrue(output.contains("$30.00"));
    }
}
